### DISCUSSION TITLE
version 2.5.0

### DISCUSSION THREAD

**Discussion by gekigek99** [gekigek99/minecraft-vanilla-server-hibernation#221] [Releases]
### major updates
- Implemented api2 in program manager
- implemented ms suspension (windows/linux/darwin)
- Added command-line arguments
- Added major ms error report to client
- Added msh console input reserved line
- Added stats query support
- Updated mshID management

### minor updates
- Added possibility to kill the minecraft server after stop command is issued
- Added notification for unofficial version
- Added msh exit command
- Added server info request for player count
- Added msh whitelist to start server (with minecraft server config import feature)
- Added msh messages in game chat
- Added check for minecraft server eula.txt file
- Added Msh server icon automatic resize
- Added minecraft server process suspension refresh to mitigate watchdog caused crash
- Added cpu/mem usage logging
- Added internet usage logging flag
- Added milliseconds in log
- Improved regex player count search
- Improved cpu percent calculation

### bugfix
- Bugfix `msh exit` ungraceful server shutdown
- Bugfix mshid generation
- Bugfix array error when bad request
- Bugfix logging of internet usage
- Bugfix bad log colors
- Bugfix ms terminal hanging
- Bugfix ms soft freeze scheduler
- Bugfix force freeze error when ms is starting
- Bugfix 0xffffffff overflow in arch386
- Bugfix sigint while ms starting does not stop ms
- Bugfix internal playercount (now connection count)
- Bugfix machine mem value overflow on 32bit arch
- Bugfix list unexpected output
- Bugfix ms command space parsing
- Bugfix query protocol related bugs
- Bugfix ms-uptime increase when ms is suspended
- Bugfix process cpu/mem on linux
- Bugfix eula.txt generation of fresh server
- Bugfix ms join timeout
- Bugfix msh args parsing
- Bugfix autorefresh prompt line
- Bugfix crash during ping emulation
- ...others

### code
- Added errco package
- Added log type
- Added log level
- Added log code
- Added tests
- Added vendor folder
- Improved logging
- Improved trace layout

**Full Changelog**: https://github.com/gekware/minecraft-server-hibernation/compare/v2.4.0...v2.5.0

<hr /><em>This discussion was created from the release <a href='https://github.com/gekware/minecraft-server-hibernation/releases/tag/v2.5.0'>version 2.5.0</a>.</em>

**Comment by A-wels** [comment]
Thank you! :)

**Comment by userh03** [comment]
This is amazing! but why doesnt it start the server when a bedrock player tries to join? I installed GeyserMC correctly and the server in hibernation only start when a Java player tries to join.
